exporters/prometheus/src/integrationTest/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServerIntegrationTest.java

```diff
@@ -0,0 +1,244 @@
+ */
+@Testcontainers(disabledWithoutDocker = true)
+class PrometheusHttpServerIntegrationTest {
+
+  private static final String COLLECTOR_IMAGE =
+      "ghcr.io/open-telemetry/opentelemetry-java/otel-collector";
+  private static final Integer COLLECTOR_HEALTH_CHECK_PORT = 13133;
+
+  private static int prometheusPort;
+  private static Resource resource;
+  private static SdkMeterProvider meterProvider;
+  private static OtlpGrpcServer grpcServer;
+  private static GenericContainer<?> collector;
+
+  @BeforeAll
+  static void beforeAll() {
+    PrometheusHttpServer prometheusHttpServer = PrometheusHttpServer.builder().setPort(0).build();
+    prometheusPort = prometheusHttpServer.getAddress().getPort();
+    resource = Resource.getDefault();
+    meterProvider =
+        SdkMeterProvider.builder()
+            .setResource(resource)
+            .registerMetricReader(prometheusHttpServer)
+            .build();
+    exposeHostPorts(prometheusPort);
+
+    grpcServer = new OtlpGrpcServer();
+    grpcServer.start();
+    exposeHostPorts(grpcServer.httpPort());
+
+    collector =
+        new GenericContainer<>(DockerImageName.parse(COLLECTOR_IMAGE))
+            .withEnv("APP_ENDPOINT", "host.testcontainers.internal:" + prometheusPort)
+            .withEnv("LOGGING_EXPORTER_VERBOSITY", "detailed")
+            .withEnv(
+                "OTLP_EXPORTER_ENDPOINT", "host.testcontainers.internal:" + grpcServer.httpPort())
+            .withClasspathResourceMapping(
+                "otel-config.yaml", "/otel-config.yaml", BindMode.READ_ONLY)
+            .withCommand("--config", "/otel-config.yaml")
+            .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("otel-collector")))
+            .withExposedPorts(COLLECTOR_HEALTH_CHECK_PORT)
+            .waitingFor(Wait.forHttp("/").forPort(COLLECTOR_HEALTH_CHECK_PORT));
+    collector.start();
+  }
+
+  @AfterAll
+  static void afterAll() {
+    meterProvider.shutdown().join(10, TimeUnit.SECONDS);
+    grpcServer.stop().join();
+    collector.stop();
+  }
+
+  @AfterEach
+  void afterEach() {
+    grpcServer.reset();
+  }
+
+  @Test
+  void endToEnd() {
+    Meter meter = meterProvider.meterBuilder("test").setInstrumentationVersion("1.0.0").build();
+
+    meter
+        .counterBuilder("requests")
+        .build()
+        .add(3, Attributes.builder().put("animal", "bear").build());
+
+    await()
+        .atMost(Duration.ofSeconds(30))
+        .untilAsserted(() -> assertThat(grpcServer.metricRequests.size()).isGreaterThan(0));
+
+    ExportMetricsServiceRequest request = grpcServer.metricRequests.get(0);
+    Assertions.assertThat(request.getResourceMetricsCount()).isEqualTo(1);
+
+    ResourceMetrics resourceMetrics = request.getResourceMetrics(0);
+    assertThat(resourceMetrics.getResource().getAttributesList())
+        .containsExactlyInAnyOrder(
+            // Resource attributes derived from the prometheus scrape config
+            stringKeyValue("service.name", "app"),
```

Comment:
It's pretty confusing that we have both `service.name` from scraping (and it contains a dot - shouldn't that be converted to an underscore?) and `service_name` from the resource. Well, nothing we can do about that in this PR.

Reply:
\`service.name` comes from `job_name` that produced the scrape in the [promtheus receiver](https://github.com/jack-berg/opentelemetry-java/blob/prometheus-compatibility/exporters/prometheus/src/integrationTest/resources/otel-config.yaml#L7) config. `service_name` comes from the attributes of the `target_info` metric from the app's prometheus exporter. 

I agree it's confusing. Best case would be for the prometheus receiver to recognize that a `target_info` is present, try to reverse the attribute key transformation based on semantic conventions (i.e. transform `service_name` back to `service.name`), and prioritize the attributes from `target_info` over those generated from the job config.